src/layers/CementLayer.ts

```diff
@@ -72,9 +84,14 @@ export class CementLayer extends WellboreBaseComponentLayer {
         }
       }
 
-      // Always add last point
+      // Always add last points
+      for (let i = c.boc - morePointsForStartAndEndMeters; i < c.boc; i += this.options.wellboreBaseComponentIncrement) {
+        const p = this.referenceSystem.project(i);
+        points.push({ point: new Point(p[0], p[1]), md: i });
+      }
```

Comment:
should validate that `i` is not negative or clamp the value when passing it into list of points.